src/vm/customattribute.cpp

```diff
@@ -294,13 +294,11 @@ HRESULT Attribute::ParseCaValue(
         IfFailGo(ca.GetU4(&len));
         pCaArg->arr.length = len;
         pCaArg->arr.pSArray = NULL;
-        if (pCaArg->arr.length == (ULONG)-1)
-            break;
 
         IfNullGo(pCaArg->arr.pSArray = pCaValueArrayFactory->Create()); 
         elementType.Init(pCaArg->type.arrayType, SERIALIZATION_TYPE_UNDEFINED, 
```

Comment:
I think it would be better to hoist this line before the original "early out if arr.Length == -1" check and leave that original check in. The code will be more maintainble if we early out - we just have to init the EnumName before we do that.